rosidl_generator_java/resource/action.java.em

```diff
@@ -76,6 +80,41 @@ import @(action_import);
 
 public class @(type_name) implements ActionDefinition {
 
+  public static class SendGoalRequest extends @(type_name)_SendGoal_Request implements GoalRequestDefinition {
+    public List<Byte> getGoalUuid() {
+      return super.getGoalId().getUuid();
+    }
```

Comment:
this comment can be safely ignored

Reply:
I think we need it. The reason for adding the `GoalRequestDefinition` and `GoalResponseDefinition` interfaces is because we have to deal with a variety of request/response definitions in a general way. E.g. [on this line in rcljava](https://github.com/osrf/ros2_java/blob/0f441c390c8b27524b87ed875192ecab9ce0df95/rcljava/src/main/java/org/ros2/rcljava/action/ActionServerImpl.java#L315) (from #53), we can't use the specific type `Fibonacci_SendGoal_Request`. Therefore, I've introduced this interface with the general methods we need.

I would rather just use the methods provided by the generated request and response definitions, but it's not possible in all cases. E.g. `@(type_name)_SendGoal_Request.getGoalId()` returns type `unique_identifier_msgs.msg.UUID`, but we can't depend on this type in `rcljava_common` because it creates a circular dependency (unique_identifier_msgs -> rosidl_generator_java -> rcljava_common -> unique_identifier_msgs).

Reply:
I suppose it's possible to modify `msg.java.em` to be "action-aware" and override these methods that are specific to actions as an alternative. I.e. detect if we're generating an action goal request/reponse message and instead of generating the typically getter/setter code, replace it with implementation for the methods defined in `GoalRequestDefinition` and `GoalResponseDefinition`. :thinking:

Reply:
> I think we need it. The reason for adding the GoalRequestDefinition and GoalResponseDefinition interfaces is because we have to deal with a variety of request/response definitions in a general way. E.g. on this line in rcljava (from #53), we can't use the specific type Fibonacci_SendGoal_Request. Therefore, I've introduced this interface with the general methods we need.

Makes sense, thanks for clarifying.
After getting used to the cpp way, I forget that type checking (partially) happens after template monomorphization and you can do things that you cannot do in java with interfaces.

> Alternatively, I suppose it's possible to modify msg.java.em to be "action-aware" and override these methods that are specific to actions. I.e. detect if we're generating an action goal request/reponse message and instead of generating the typically getter/setter code, replace it with implementation for the methods defined in GoalRequestDefinition and GoalResponseDefinition.

I think that the current approach is good :+1: